src/main/java/at/codersbay/datastructures/Fractional.java

```diff
@@ -5,30 +5,69 @@
  * e.g.: 1/2, 1/3, 24/37, ..
  */
 public class Fractional {
+	
+	private final Integer numerator;
+	private final Integer denominator;
 
-    public Fractional(Integer numerator, Integer denominator) {
-        // TODO implement to fix tests in FractionalTest1
+    public Fractional(Integer numerator, Integer denominator) throws IllegalArgumentException {
+    	if (numerator == null || denominator == null) {
+    		throw new IllegalArgumentException("You shall not pass null!");
+    	} else if (denominator == 0) {
+			throw new IllegalArgumentException("You shall not divide through zero!");
+		} else {
+			this.numerator = numerator;
+			this.denominator = denominator;
+		}
     }
 
     public Integer getNumerator() {
-        // TODO implement to fix tests in FractionalTest1
-        return -1;
+        return numerator;
     }
 
     public Integer getDenominator() {
-        // TODO implement to fix tests in FractionalTest1
-        return -1;
+        return denominator;
     }
 
     public Float asFloat() {
-        // TODO implement to fix tests in FractionalTest2
-        return -1f;
+    	Float result = ((float) this.numerator / (float) this.denominator);
+        return result;
     }
 
     @Override
-    public boolean equals(Object obj) {
-        // TODO implement to fix tests in FractionalTest3
-        return false;
+	public boolean equals(Object obj) {
+		// TODO handle cases where a non-fractional is passed as an argument
+		Fractional other = (Fractional) obj;
+		boolean result = false;
+		int numerator1 = this.getNumerator();
+		int numerator2 = other.getNumerator();
+		int denominator1 = this.getDenominator();
+		int denominator2 = other.getDenominator();
+		if (numerator1 == 0 && numerator2 == 0) {
+			result = true;
+		} else if (numerator1 == numerator2 && denominator1 == denominator2) {
+			result = true;
+		} else {
+			int commonDivisor1 = greatestCommonDivisor(numerator1, denominator1);
+			int commonDivisor2 = greatestCommonDivisor(numerator2, denominator2);
+			numerator1 /= commonDivisor1;
+			denominator1 /= commonDivisor1;
+			numerator2 /= commonDivisor2;
+			denominator2 /= commonDivisor2;
+			if (numerator1 == numerator2 && denominator1 == denominator2) {
+				result = true;
+			}
+		}
+		return result;
+	}
+    
+    private int greatestCommonDivisor(int numerator, int denominator) {
+    	int result = 1;
```

Comment:
no need to give result an inital value I think.. also if it is a value that's not really correct but just an initial value I like to use `-1` as a starting point to indicate an invalid value